### ISSUE TITLE
Add new shader rotation algorithm

### ISSUE THREAD

**Issue by dacap** [aseprite/aseprite#4331]
With the [help of torcado](https://twitter.com/torcado/status/1760701389796200948) we would like to add their cleanEdge rotation algorithm based on [this shaders](https://gist.github.com/torcado194/e2794f5a4b22049ac0a41f972d14c329).

**Comment by gui-marc** [comment]
Could this be implemented in the current state of the application? Did some research in the app's code but based on https://github.com/aseprite/aseprite/issues/3516 i guess Aseprite currently cannot use shaders for rendering images. 

Am I right or did I miss something?

**Comment by dacap** [comment]
There are still some work to do. For #3516 we have already something working in the `ShaderRenderer` class, and for this issue I have a local stash with some code, but they are two independent features (one uses shaders to render sprites, the other uses shaders to modify the sprite).

I'll see if I can push something about this in a draft PR with my local changes.

**Comment by gui-marc** [comment]
In fact they are two independent features, but can the current `ShaderRenderer` use shaders to modifty the sprite? I have some interest and experience in CG stuff and would love to help implement that :)

**Comment by dacap** [comment]
> can the current ShaderRenderer use shaders to modifty the sprite? 

No, the purpose of the `ShaderRenderer` is to render the sprite. If you want to create a shader to modify the sprite you have to use the `app::make_shader()` function defined in `shader_helpers.h` with your own shader/inputs/outputs (I have some code related to this for this feature, not yet public).

**Comment by gui-marc** [comment]
>  (I have some code related to this for this feature, not yet public)

So this feature is almost fully implemented, right? There's any other related issue related to this you think I could help?

**Comment by dacap** [comment]
No, it's not fully implemented (only on RGB, broken code, the shader wasn't ready, and we required to get the proper transformation matrix yet, and too little time to debug code yet).

Probably you are looking for something much simpler, like just running a shader in a rectangular area (like an FX). You are free to create a new issue to request a new feature in that case.
